scripts/generate_records/generate_db_records.js

```diff
@@ -0,0 +1,439 @@
+ * along with files per granule and granuleExecutions
+ *
+ * @param {Knex} knex
+ * @param {string} providerId
+ * @param {CollectionDetails} collection
+ * @param {number} numberOfGranules
+ * @param {number} filesPerGranule
+ * @param {number} granulesPerBatch
+ * @param {number} executionsPerBatch
+ * @param {number} concurrency
+ * @param {boolean} variance
+ * @param {boolean} swallowErrors
+ * @returns {Promise<void>}
+ */
+
+const uploadDBGranules = async (
+  knex,
+  providerId,
+  collection,
+  numberOfGranules,
+  filesPerGranule,
+  granulesPerBatch,
+  executionsPerBatch,
+  concurrency,
+  variance = false,
+  swallowErrors = false
+) => {
+  const collectionPgModel = new CollectionPgModel();
+  const providerPgModel = new ProviderPgModel();
+  const dbCollection = await collectionPgModel.get(
+    knex,
+    { name: collection.name, version: collection.version }
+  );
+  const dbProvider = await providerPgModel.get(knex, { name: providerId });
+  const collectionCumulusId = dbCollection.cumulus_id;
+  const providerCumulusId = dbProvider.cumulus_id;
+  const models = {
+    geModel: new GranulesExecutionsPgModel(),
+    executionModel: new ExecutionPgModel(),
+    granuleModel: new GranulePgModel(),
+    fileModel: new FilePgModel(),
+  };
+  const iterableParamGenerator = getBatchParamGenerator({
+    knex,
+    numberOfGranules,
+    collectionCumulusId,
+    providerCumulusId,
+    filesPerGranule,
+    granulesPerBatch,
+    executionsPerBatch,
+    models,
+    variance,
+    swallowErrors,
+  });
+  await pMap(
+    iterableParamGenerator,
+    // this lambda function swallows uploadDataBatch's return to prevent ballooning memory use
+    async (params) => {
+      await uploadDataBatch(params);
+    },
+    { concurrency }
+  );
+};
+
+/**
+ * Parse executions per batch and granules per batch based on a given ratio
+ *
+ * @param {string} executionsPerGranule - executionsPerGranule in <executions>:<granules>
+ * @returns {{executionsPerBatch: number, granulesPerBatch: number}}
+ */
+const parseExecutionsGranulesBatch = (executionsPerGranule) => {
+  // expect to come in format 'x:y'
+  try {
+    const split = executionsPerGranule.split(':');
+    if (split.length < 2) {
+      throw new Error(`only 1 value could be split from ${executionsPerGranule}`);
+    }
+    const executionsPerBatch = Number.parseInt(split[0], 10);
+    const granulesPerBatch = Number.parseInt(split[1], 10);
+    return { executionsPerBatch, granulesPerBatch };
+  } catch (error) {
+    throw new Error(`cannot parse ${executionsPerGranule}, expected format <executions>:<granules> ratio \n${error}`);
+  }
+};
+
+/**
+ * parse command line args for run parameters
+ *
+ * @returns {{
+ *   granules: number,
+ *   files: number,
+ *   granulesPerBatch: number
+ *   executionsPerBatch: number
+ *   collections: number
+ *   concurrency: number
+ *   variance: boolean
+ *   swallowErrors: boolean
+ * }}
+ */
+const parseArgs = () => {
```

Comment:
two things I thought about after our discussion on the options/args:
1. what are you doing to handle any options passed in that don't match the ones available?
2. to handle the close but not quite there options for things like executions_per_granules that you originally had, maybe you can grab those and give an error message like "did you mean executionsPerGranule?"

neither of these are completely necessary, maybe nice to have. Just something that came to mind after our discussion.

Reply:
I think because this is a developer tool I'm not gonna think too hard about it

Reply:
totally fair, just wanted to share what I thought of